src/ILVerify/src/ILImporter.Verify.cs

```diff
@@ -1267,7 +1267,7 @@ void ImportLoadIndirect(TypeDesc type)
             var address = Pop();
 
             if (type == null)
-                type = GetWellKnownType(WellKnownType.Object);
+                type = address.Type;
```

Comment:
Check that the type is objectref? `ldind.ref` with `int&` on the stack should be invalid, I think.